packages/stark-ui/src/modules/pretty-print/components/pretty-print.component.ts

```diff
@@ -154,9 +158,15 @@ export class StarkPrettyPrintComponent extends AbstractStarkUiComponent implemen
 					case "html":
 						prismGrammar = Prism.languages.markup;
 						prismClass = prismClassPrefix + "markup";
-						// FIXME Adapt following line after replacing "pretty-data" dependency. See: https://github.com/NationalBankBelgium/stark/issues/2543
-						// this.prettyString = prettyData.xml(this.data);
-						this.prettyString = this.data;
+						if (this.format === "xml") {
```

Comment:
I suggest to move this block to the ´case "xml"´ above 😉

Reply:
I was hesitating because the logic for highlighting is the same. But you're right, it will be better for the understanding to split this since the formatter logic is not the same anymore 😊